src/ListenerProviderInterface.php

```diff
@@ -11,9 +13,9 @@ interface ListenerProviderInterface
     /**
      * @param object $event
      *   An event for which to return the relevant listeners.
-     * @return iterable[callable]
-     *   An iterable (array, iterator, or generator) of callables.  Each
+     * @return Traversable[callable]
+     *   A list of callables. Each
      *   callable MUST be type-compatible with $event.
      */
-    public function getListenersForEvent(object $event) : iterable;
+    public function getListenersForEvent($event) : Traversable;
```

Comment:
Arrays do not satisfy `Traversable`, so this fork is strictly speaking not API-compatible. I suggest removing the return type hint altogether.

Reply:
Arrays do not satisfy `Traversable`, but `Traversable` satisfies `iterable`. Because earlier PHP versions do not have the `iterable` type, I had to choose between `array` and `Traversable`. I chose `Traversable` because it allows for more flexibility.

The premise for forward-compatibility here is that all implementations of this fork would still work just the same if they suddenly begun to also implement the original PSR-14 interfaces. Doesn't this interface satisfy that condition?

Reply:
In that case, yes. If FC is what's important then `Traversable` is the logical choice.

But in the event of porting an implementation over to this fork, `array` returns would need to be wrapped. It would probably be a good idea to make a list somewhere, like in the README, that outlines what changes would need to be made to an implementation to port it over to this forked standard.

Reply:
That's true. But I haven't considered that someone may have a need to port PSR-14 implementations to this. I thought that mostly people would be porting from this to PSR-14 :D. But you're right!

Reply:
Done.